test/inductor/test_codecache.py

```diff
@@ -152,6 +155,121 @@ def test_cache_load_model(self, device, dtype):
+        """
+        Test caching the same graph, but under conditions that introduce guards
+        for tensor sizes < int32.
+        """
+        if device == "cuda" and not HAS_CUDA:
+            raise unittest.SkipTest("requires CUDA")
+        if device == "cuda" and dtype == torch.bfloat16 and not SM80OrLater:
+            raise unittest.SkipTest("requires SM80 or later")
+
+        def fn(x, y):
+            return (x + x, y + y)
+
+        compiled_fn = torch.compile(fn)
+
+        # Mark all tensor arg dimensions as dynamic to cause all shapes
+        # to be symbolic
+        def rand_dynamic(*args):
+            t = torch.rand(*args, device=device, dtype=dtype)
+            for i in range(len(args)):
+                torch._dynamo.mark_dynamic(t, i)
+            return t
+
+        # 1) Both args' shapes are small. We expect guards testing that
+        # both do not exceed int32.
+        a = rand_dynamic(5, 5)
+        b = rand_dynamic(7, 7)
+
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 1)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 0)
+
+        # A second call should hit.
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 1)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 1)
+
+        # 2) One arg's shape is small; the other is large. We expect a
+        # new guard expression that only tests that the first does not
+        # exceed int32.
+        a = rand_dynamic(7, 7)
+        b = rand_dynamic(47000, 47000)
+
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 2)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 1)
+
+        # A second call should hit.
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 2)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 2)
+
+        # 3) The other arg's shape is large. We expect yet another
+        # guard expression that only tests that the second does not
+        # exceed int32.
+        a = rand_dynamic(47000, 47000)
+        b = rand_dynamic(7, 7)
+
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 3)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 2)
+
+        # A second call should hit.
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 3)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 3)
```

Comment:
So I'm thinking about the test you wrote here, and it seems to me that many pre-existing tests that are checking cache behavior could be repurposed for the FX graph cache. The general idea is that as long as we aren't testing for guards produced by Dynamo, the test can be run in two ways: once without clearing the in memory code cache, and once clearing the in memory code cache, so you hit the FX graph cache. Something to think about.

Reply:
There is no in-memory FX graph cache in the implementation so far though. It's only on disk. Do you mean clearing on the on-disk cache? (btw, I have a TODO to investigate whether extending to an in-memory cache too would help compile time, but my guess is that it will be in the noise)

Reply:
The "in-memory" cache is the Dynamo compiled code cache